From: Florian Westphal <fw@strlen.de>
Date: Tue, 3 May 2016 16:31:00 +0200
Subject: netdevice: add helper to update trans_start
Patch-mainline: v4.7-rc1
Git-commit: ba162f8eed61a7e71e26455ce1cff5b5898a3579
References: bsc#1024349 FATE#321240 bsc#1024355 FATE#321241

trans_start exists twice:
- as member of net_device (legacy)
- as member of netdev_queue

In order to get rid of the legacy case, add a helper for the
dev->trans_update (this patch), then convert spots that do

dev->trans_start = jiffies

to use this helper (next patch).

This would then allow us to change the helper so that it updates the
trans_stamp of netdev queue 0 instead.

Signed-off-by: Florian Westphal <fw@strlen.de>
Signed-off-by: David S. Miller <davem@davemloft.net>
Acked-by: Benjamin Poirier <bpoirier@suse.com>
---
 include/linux/netdevice.h |    6 ++++++
 1 file changed, 6 insertions(+)

--- a/include/linux/netdevice.h
+++ b/include/linux/netdevice.h
@@ -3402,6 +3402,12 @@ static inline void txq_trans_update(stru
 		txq->trans_start = jiffies;
 }
 
+/* legacy drivers only, netdev_start_xmit() sets txq->trans_start */
+static inline void netif_trans_update(struct net_device *dev)
+{
+	dev->trans_start = jiffies;
+}
+
 /**
  *	netif_tx_lock - grab network device transmit lock
  *	@dev: network device
